server/integration/endpoint/src/main/java/org/infinispan/server/endpoint/deployments/AbstractCacheStoreExtensionProcessor.java

```diff
@@ -0,0 +1,52 @@
+package org.infinispan.server.endpoint.deployments;
+
+import org.infinispan.server.endpoint.Constants;
+import org.infinispan.server.endpoint.subsystem.ExtensionManagerService;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.msc.service.Service;
+import org.jboss.msc.service.ServiceBuilder;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceName;
+import org.jboss.msc.value.InjectedValue;
+
+import static org.infinispan.server.endpoint.EndpointLogger.ROOT_LOGGER;
+
+public abstract class AbstractCacheStoreExtensionProcessor<T> extends AbstractServerExtensionProcessor<T> {
+
```

Comment:
This is precisely the reason why `org.infinispan.server.endpoint.deployments.ServerExtensionDependenciesProcessor` class exists. It's job is to add the Infinispan dependency to the deployed jars. The reason this is done in a different processor class is because this class gets executed at a different phase, the phase of adding dependencies. Look for references to this class and you'll see what you mean.

So, what you need to do is tweak `ServerExtensionDependenciesProcessor.hasInfinispanExtensions` to check for any new cache store/writer service implementations in deployed jars and if found add the infinispan dependency.